classlib/src/main/java/org/teavm/classlib/java/util/TList.java

```diff
@@ -50,6 +50,94 @@ default void sort(TComparator<? super E> c) {
         TCollections.sort(this, c);
     }
 
+    @Override
+    default void addFirst(E e) {
+        add(0, e);
+    }
+
+    @Override
+    default void addLast(E e) {
+        add(e);
+    }
+
+    @Override
+    default E getFirst() {
+        if (isEmpty()) {
+            throw new TNoSuchElementException();
+        }
+        return get(0);
+    }
+
+    @Override
+    default E getLast() {
+        if (isEmpty()) {
+            throw new TNoSuchElementException();
+        }
+        return get(this.size() - 1);
+    }
+
+    @Override
+    default E removeFirst() {
+        if (isEmpty()) {
+            throw new TNoSuchElementException();
+        }
+        return remove(0);
+    }
+
+    @Override
+    default E removeLast() {
+        if (isEmpty()) {
+            throw new TNoSuchElementException();
+        }
+        return remove(this.size() - 1);
+    }
+
+    @Override
+    default TList<E> reversed() {
+        return new TAbstractList<>() {
```

Comment:
Other methods worth implementing here (only for performance reasons): addAll, clear, isEmpty, contains, addFirst, addLast, getFirst, getLast, removeFirst, removeLast, reversed, listIterator